web/src/stream_data.ts

```diff
@@ -465,6 +465,14 @@ export function get_color(stream_id: number): string {
     return sub.color;
 }
 
+export function get_color_by_stream_name(stream_name: string): string {
+    const sub = get_sub_by_name(stream_name);
+    if (sub === undefined) {
+        return DEFAULT_COLOR;
+    }
+    return sub.color;
+}
```

Comment:
I think you should be able to use `get_color`; edit history entries I believe have access to actual stream IDs, not just names.